packages/flutter_tools/bin/xcode_backend.sh

```diff
@@ -46,37 +46,41 @@ BuildApp() {
     target_path="${FLUTTER_TARGET}"
   fi
 
+  # Use FLUTTER_BUILD_MODE if it's set, otherwise use the Xcode build configuration name
+  # This means that if someone wants to use an Xcode build config other than Debug/Profile/Release,
+  # they _must_ set FLUTTER_BUILD_MODE so we know what type of artifact to build.
+  local build_mode="$(echo "${FLUTTER_BUILD_MODE:-${CONFIGURATION}}" | tr "[:upper:]" "[:lower:]")"
```

Comment:
This seems really implicit and would cause issues with flavors https://medium.com/@salvatoregiordanoo/flavoring-flutter-392aaa875f36. 

While you're here, I think it would be great if the flavors mechanism could be documented somewhere in https://flutter.io/docs/ if priorities could work. (So you can append to that doc the fact that you if create custom schemes / build configs, a $FLUTTER_BUILD_MODE variable needs to be defined or passed in)